core/unix/os.c

```diff
@@ -4454,17 +4454,29 @@ fd_priv_dup(file_t curfd)
          * so how do we tell if the flag is supported?  try calling once at init?
          */
         newfd = fcntl_syscall(curfd, F_DUPFD, min_dr_fd);
-        if (newfd < 0) {
+        int try_fd = min_dr_fd / 2;
```

Comment:
Maybe remind the reader that F_DUPFD returns the lowest numbered descriptor available >= try_fd, and we want to keep DR's file descriptors towards the top to minimize possibility for transparency issues.